datalake/src/main/java/com/sequenceiq/datalake/flow/dr/validation/DatalakeBackupValidationActions.java

```diff
@@ -0,0 +1,174 @@
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.action.Action;
+
+import com.dyngr.exception.PollerStoppedException;
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.datalakedr.model.DatalakeBackupStatusResponse;
+import com.sequenceiq.cloudbreak.event.ResourceEvent;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.flow.SdxContext;
+import com.sequenceiq.datalake.flow.SdxEvent;
+import com.sequenceiq.datalake.flow.dr.backup.DatalakeBackupFailureReason;
+import com.sequenceiq.datalake.flow.dr.validation.event.DatalakeBackupValidationFailedEvent;
+import com.sequenceiq.datalake.flow.dr.validation.event.DatalakeBackupValidationWaitRequest;
+import com.sequenceiq.datalake.flow.dr.validation.event.DatalakeTriggerBackupValidationEvent;
+import com.sequenceiq.datalake.metric.MetricType;
+import com.sequenceiq.datalake.metric.SdxMetricService;
+import com.sequenceiq.datalake.service.AbstractSdxAction;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.dr.SdxBackupRestoreService;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.core.Flow;
+
+@Configuration
+public class DatalakeBackupValidationActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatalakeBackupValidationActions.class);
+
+    private static final String BACKUP_ID = "BACKUP-ID";
+
+    private static final String REASON = "REASON";
+
+    @Inject
+    private SdxBackupRestoreService sdxBackupRestoreService;
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Inject
+    private SdxMetricService metricService;
+
+    @Inject
+    private SdxService sdxService;
+
+    @Bean(name = "DATALAKE_TRIGGERING_BACKUP_VALIDATION_STATE")
+    public Action<?, ?> triggerDatalakeBackupValidationAction() {
+        return new AbstractSdxAction<>(DatalakeTriggerBackupValidationEvent.class) {
+            @Override
+            protected void doExecute(SdxContext context, DatalakeTriggerBackupValidationEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Triggering datalake backup validation for {}", payload.getResourceId());
+                variables.put(REASON, payload.getReason().name());
+
+                SdxCluster sdxCluster = sdxService.getById(payload.getResourceId());
+                sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.RUNNING,
```

Comment:
Is it intentional to set RUNNING status?

Reply:
Yes, the idea is that the validation, as well as other backup operations, do not affect the DL as a whole but simply serve as background operations that are occurring while the DL is still up and running.